🐛 BUG: `[alias]` no longer working in `wrangler dev` in v3.80.0

### Which Cloudflare product(s) does this pertain to?

Wrangler

### What version(s) of the tool(s) are you using?

3.80.0

### What version of Node are you using?

_No response_

### What operating system and version are you using?

Mac Sonoma 14.6.1

### Describe the Bug

The following works fine in v3.79.0:
```toml
[alias]
"critters" = "./my-shim.mjs"
"@opentelemetry/api" = "./my-shim.mjs"
```

But in v3.80.0 it errors in `wrangler dev` as if the aliases have no effect:

```
✘ [ERROR] Could not resolve "critters"
# etc...
```

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

--- a/.changeset/grumpy-apples-juggle.md
+++ b/.changeset/grumpy-apples-juggle.md
@@ -1,0 +1,7 @@
+---
+"wrangler": patch
+---
+
+fix: ensure that `alias` config gets passed through to the bundler when using new `--x-dev-env`
+
+Fixes #6898

--- a/packages/wrangler/src/__tests__/api/startDevWorker/ConfigController.test.ts
+++ b/packages/wrangler/src/__tests__/api/startDevWorker/ConfigController.test.ts
@@ -123,14 +123,20 @@ describe("ConfigController", () => {
 			dev: {
 				origin: { hostname: "myexample.com" },
 			},
+			build: {
+				alias: { foo: "bar" },
+			},
 		});
-		// expect `dev` field to be overwritten and all other config to remain intact
+		// expect `dev` and `build.alias` fields to be overwritten and all other config to remain intact
 		await expect(event3).resolves.toMatchObject({
 			type: "configUpdate",
 			config: {
 				entrypoint: path.join(process.cwd(), "src/index.ts"),
 				directory: process.cwd(),
 				build: {
+					alias: {
+						foo: "bar",
+					},
 					additionalModules: [],
 					define: {},
 					format: "modules",

--- a/packages/wrangler/src/api/startDevWorker/ConfigController.ts
+++ b/packages/wrangler/src/api/startDevWorker/ConfigController.ts
@@ -257,6 +257,7 @@ async function resolveConfig(
 		triggers: await resolveTriggers(config, input),
 		env: input.env,
 		build: {
+			alias: input.build?.alias ?? config.alias,
 			additionalModules: input.build?.additionalModules ?? [],
 			processEntrypoint: Boolean(input.build?.processEntrypoint),
 			bundle: input.build?.bundle ?? !config.no_bundle,

